### ISSUE TITLE
a11y[vpat 169]: add missing role to cards and improve aria-labels

### ISSUE THREAD

**Issue by aalves08** [rancher/dashboard#15881]
**Setup**
- Rancher version:
- Rancher UI Extensions:
- Browser type & version:

**Describe the bug**
add missing role to cards and improve aria-labels in pages that use `SelectIconGrid`:
- Cluster management create
- Auth providers selection
- navlinks

Double-check and fix (if needed) cards in:
- Charts
- Secret creation

**To Reproduce**

**Result**

**Expected Result**

**Screenshots**

**Additional context**
